File: binarygap.c
// you can write to stdout for debugging purposes, e.g.
// printf("this is a debug message\n");

int main() {int solution(int N) {
	solution(10);
}

int solution(int N) {
    int gap = 0;
    int max = 0;
    int start = 0;
    while (N > 0) {
        if (N & 0x1) {
            if (start) {
                max = (max > gap) ? max : gap;
            }
            else {
                start = 1;
            }
            gap = 0;
        }
        else {
            gap++;
        }

        N >>= 1;
    }
    return(max);
}

File: src/Codility/BinaryGap.java
package Codility;

public class BinaryGap {
	public static void main (String[] args) {
		System.out.println("Solution=" + solution(10));
	}
	
    public static int solution(int N) {
        int gap = 0;
        int bg = 0;
        boolean start = false;
        while (N > 0) {
            if ((N & 0x1) == 0) {
                if (start) gap++;
            }
            else {
                if (!start) {
                    start = true;
                }
                else {
                    bg = (gap > bg) ? gap : bg;
                    gap = 0;
                }
            }
            N >>= 1;
        }
        return(bg);
    }
}